erpnext/accounts/general_ledger.py

```diff
@@ -124,6 +124,8 @@ def make_entry(args, adv_adj, update_outstanding, from_repost=False):
 	gle.insert()
 	gle.run_method("on_update_with_args", adv_adj, update_outstanding, from_repost)
 	gle.submit()
+	frappe.db.sql("""UPDATE `tabGL Entry` SET cost_center = %s WHERE name = %s """, (args.cost_center, gle.name))
```

Comment:
This is very wrong. Pass cost center in GL entries itself. No need of explicit commit